crates/kitsune_p2p/types/src/config.rs

```diff
@@ -396,22 +376,53 @@ pub struct KitsuneP2pConfig {
     #[serde(default)]
     pub tuning_params: KitsuneP2pTuningParams,
 
-    /// The network used for connecting to other peers
-    pub network_type: NetworkType,
-
     /// All tracing logs from kitsune tasks will be instrumented to contain this string,
     /// so that logs from multiple instances in the same process can be disambiguated.
     #[serde(default)]
     pub tracing_scope: Option<String>,
 }
 
+#[cfg(feature = "test_utils")]
 impl Default for KitsuneP2pConfig {
     fn default() -> Self {
+        Self::mem()
+    }
+}
+
+impl KitsuneP2pConfig {
+    /// Minimal but non-functional config. Without a transport pool set,
+    /// nothing will work.
+    pub fn mem() -> Self {
         Self {
-            transport_pool: Vec::new(),
+            transport_pool: vec![TransportConfig::Mem {}],
+            bootstrap_service: None,
+            tuning_params: KitsuneP2pTuningParams::default(),
+            tracing_scope: None,
+        }
+    }
+
+    pub fn testing() -> Self {
+        Self {
+            transport_pool: vec![TransportConfig::WebRTC {
+                signal_url: "wss://signal.holo.host".to_string(),
+                webrtc_config: None,
+            }],
+            bootstrap_service: Some(Url2::parse("https://bootstrap.holo.host")),
+            tuning_params: KitsuneP2pTuningParams::default(),
+            tracing_scope: None,
+        }
+    }
```

Comment:
- These are holo addresses, they shouldn't be in any kitsune crate.
- These are production addresses, they shouldn't be in something marked 'testing'.

Reply:
Good points. Unfortunately the struct is hard to construct properly because [it's marked non_exhaustive](https://github.com/holochain/holochain/blob/63df467df4e5ee675abf33fe301e24ee267b6535/crates/kitsune_p2p/types/src/config.rs#L363), so you can't just construct it from parts. I'm not sure why it's marked as such, but if we removed the non_exhaustive, I could just plop this struct in the two places that `testing()` is used and remove the fn. If we don't remove non_exhaustive, it's trickier (default is mem, and then we override it with non-mem config?)

Reply:
I removed `non_exhaustive` because it seems like the right thing to do given what is needed.

@mattyg I think you added this, if you want to make a case for keeping it, I'll put it back in.

Reply:
Okay, but you missed the second bullet point. We still don't want to hit production services in test code.

Reply:
What's the viable alternative, when multi conductors are involved?

Reply:
If the multiple conductors are in the same process, use rendezvous. If they are in multiple processes, use hc_run_local_services.

Reply:
I'll try to get that done today

Reply:
The only places which use this config:

1. [sandbox generation](https://github.com/holochain/holochain/blob/67a8d6e95814b4025cff1c3c3e5eb796e77a7895/crates/hc_sandbox/src/generate.rs#L36)
2. [these tests](https://github.com/holochain/holochain/blob/67a8d6e95814b4025cff1c3c3e5eb796e77a7895/crates/holochain/tests/tests/test_utils/mod.rs#L442)

I think it's valid to use the production infra for 1, but not 2. Does that sound right?

Reply:
Well, maybe even sandbox generation is not a valid place. But the sandbox needs some serious attention anyway, so I'm not inclined to fix this now. The other case is easy to fix though, they're all single-conductor tests which can use mem.

Reply:
issue: https://github.com/holochain/holochain/issues/4407